miden/src/cli/data.rs

```diff
@@ -27,20 +27,43 @@ impl InputFile {
 
         println!("Reading input file `{}`", path.display());
 
-        // read input file to string
-        let inputs_file = fs::read_to_string(&path)
-            .map_err(|err| format!("Failed to open input file `{}` - {}", path.display(), err))?;
-
-        // deserialize input data
-        let inputs: InputFile = serde_json::from_str(&inputs_file)
-            .map_err(|err| format!("Failed to deserialize input data - {}", err))?;
-
-        Ok(inputs)
+        // read input file to string.
+        match fs::read_to_string(&path) {
```

Comment:
I think this works but the `println!()` above is somewhat misleading as when the input file was not specified and does not exist, we are actually not reading anything. I'd prefer refactoring so that "Reading input file..." is printed out only when relevant. Specifically, the overall logic could be:

* If the `input_path` is not provided, check if `program_path.with_extension("inputs")` exists.
  * If it doesn't, return with blank inputs.
  * If it does, update the path as you do now.
* Attempt to read from `input_path` and return w/e was read.